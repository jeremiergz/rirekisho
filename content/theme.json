{
  "colors": {
    "background": "white",
    "body1": "#262626",
    "body2": "#575757",
    "caption": "#808080",
    "dark": "#212121",
    "error": "#f44336",
    "info": "#2196f3",
    "primary": "#17365c",
    "secondary": "#31859a",
    "subtitle": "#d3d3d3",
    "success": "#4caf50",
    "text": "#262626",
    "white": "#ffffff"
  },
  "fonts": {
    "main": "'Open Sans', sans-serif",
    "title": "'Baloo Bhaina 2', cursive"
  },
  "fontWeights": {
    "bold": 800,
    "bolder": 600,
    "regular": 400
  }
}
