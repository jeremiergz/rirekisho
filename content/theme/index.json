{
  "colors": {
    "dark": {
      "background": "#212121",
      "body1": "#d3d3d3",
      "body2": "#808080",
      "caption": "#d3d3d3",
      "dark": "#212121",
      "error": "#f44336",
      "info": "#2196f3",
      "primary": "#265a99",
      "secondary": "#50b0c7",
      "subtitle": "#575757",
      "success": "#4caf50",
      "text": "#d3d3d3",
      "white": "#ffffff"
    },
    "light": {
      "background": "white",
      "body1": "#262626",
      "body2": "#575757",
      "caption": "#808080",
      "dark": "#212121",
      "error": "#f44336",
      "info": "#2196f3",
      "primary": "#17365c",
      "secondary": "#31859a",
      "subtitle": "#d3d3d3",
      "success": "#4caf50",
      "text": "#262626",
      "white": "#ffffff"
    }
  },
  "fonts": {
    "main": "'Open Sans', sans-serif",
    "title": "'Baloo Bhaina 2', cursive"
  },
  "fontWeights": {
    "bold": 800,
    "bolder": 600,
    "regular": 400
  }
}
