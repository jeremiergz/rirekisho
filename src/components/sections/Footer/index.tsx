import Anchor from '@/components/common/Anchor';
import useFooterData from '@/components/hooks/data/useFooterData';
import { useTheme } from '@/components/providers/ThemeProvider';
import HeartIcon from '@/components/svgs/icons/Heart';
import clsx from 'clsx';
import { GatsbyImage, getImage } from 'gatsby-plugin-image';
import React from 'react';

function Footer(): JSX.Element {
  const {
    gatsbyjs,
    gatsbyjsDark,
    siteMetadata: { name, repositoryURL, version },
    tailwindcss,
    tailwindcssDark,
  } = useFooterData();
  const { mode } = useTheme();

  const appDetails = `${name.toLowerCase()} v${version}`;
  const sourceCodeURL = `${repositoryURL}/tree/${version}`;

  return (
    <footer
      className={clsx(
        'flex flex-col h-40 md:h-32 items-center justify-center mt-4 relative text-white',
        'bg-gradient-to-b from-secondary to-primary dark:from-secondary-dark dark:via-primary-dark dark:to-primary',
        'print:hidden',
      )}
    >
      <div className="flex xs:flex-col md:flex-row items-center justify-center xs:mb-4 md:mb-4">
        <div className="flex items-center xs:mb-2 md:mb-0 xs:mx-0 md:mx-4">
          <span className="font-semibold mr-1 text-sm text-white dark:text-primary">Made with</span>
          <HeartIcon height={14} width={14} />
          <span className="font-semibold ml-1 text-sm text-white dark:text-primary">using</span>
        </div>
        <div>
          <Anchor className="mr-2" external href="https://gatsbyjs.com">
            <GatsbyImage
              alt="Go to gatsbyjs.com"
              className="h-14 w-14"
              image={getImage(mode === 'light' ? gatsbyjs.childImageSharp : gatsbyjsDark.childImageSharp)}
            />
          </Anchor>
          <Anchor className="ml-2" external href="https://tailwindcss.com">
            <GatsbyImage
              alt="Go to tailwindcss.com"
              className="h-14 w-14"
              image={getImage(mode === 'light' ? tailwindcss.childImageSharp : tailwindcssDark.childImageSharp)}
            />
          </Anchor>
        </div>
      </div>
      <div className="flex flex-col items-center justify-center">
        <span className="font-semibold text-xs text-gray-300 dark:text-gray-300">{appDetails}</span>
        <Anchor className="leading-3" external href={sourceCodeURL}>
          <span className="border-b border-dotted border-gray-400 text-2xs text-gray-400">source code</span>
        </Anchor>
      </div>
    </footer>
  );
}

export default Footer;
