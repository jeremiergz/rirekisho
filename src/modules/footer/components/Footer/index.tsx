import Anchor from '@/common/components/misc/Anchor';
import HeartIcon from '@/common/components/svgs/icons/Heart';
import useSiteMetadata from '@/common/hooks/useSiteMetadata';
import clsx from 'clsx';
import { GatsbyImage, getImage } from 'gatsby-plugin-image';
import useFooterData from '../../hooks/useFooterData';

function Footer(): JSX.Element {
  const { gatsbyjs, tailwindcss } = useFooterData();
  const { name, repositoryURL, version } = useSiteMetadata();

  const appDetails = `${name.toLowerCase()} v${version}`;
  const sourceCodeURL = `${repositoryURL}/tree/${version}`;

  return (
    <footer
      className={clsx(
        'relative mt-4 flex h-40 flex-col items-center justify-center bg-primary text-gray-100 md:h-32',
        'print:hidden',
      )}
    >
      <div className="flex items-center justify-center xs:mb-4 xs:flex-col md:mb-4 md:flex-row">
        <div className="flex items-center xs:mx-0 xs:mb-2 md:mx-4 md:mb-0">
          <span className="mr-1 text-sm font-semibold text-gray-300">Made with</span>
          <HeartIcon height={14} width={14} />
          <span className="ml-1 text-sm font-semibold text-gray-300">using</span>
        </div>
        <div>
          <Anchor className="mr-2" external href="https://gatsbyjs.com">
            <GatsbyImage alt="Go to gatsbyjs.com" className="h-14 w-14" image={getImage(gatsbyjs.childImageSharp)} />
          </Anchor>
          <Anchor className="ml-2" external href="https://tailwindcss.com">
            <GatsbyImage
              alt="Go to tailwindcss.com"
              className="h-14 w-14"
              image={getImage(tailwindcss.childImageSharp)}
            />
          </Anchor>
        </div>
      </div>
      <div className="flex flex-col items-center justify-center">
        <span className="text-xs font-semibold text-slate-900">{appDetails}</span>
        <Anchor className="leading-3" external href={sourceCodeURL}>
          <span className="border-b border-dotted border-gray-900 text-2xs text-gray-900">source code</span>
        </Anchor>
      </div>
    </footer>
  );
}

export default Footer;
