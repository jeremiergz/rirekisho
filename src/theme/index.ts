const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'] as const;
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const colors = {
  background: 'white',
  dark: '#212121',
  error: '#f44336',
  info: '#2196f3',
  primary: '#17365c',
  secondary: '#31859a',
  success: '#4caf50',
  text: '#262626',
  white: '#ffffff',
} as const;

const fonts = {
  main: "'Open Sans', sans-serif",
  title: "'Baloo Bhaina 2', cursive",
} as const;

const fontWeights = {
  bold: 800,
  bolder: 600,
  regular: 400,
} as const;

const theme = {
  breakpoints,
  colors,
  fonts,
  fontWeights,
} as const;

type Theme = typeof theme;

export type { Theme };
export default theme;
