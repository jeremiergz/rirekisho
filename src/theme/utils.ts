/**
 * Shades a color by a percentage and returns changed color as an hexadecimal color.
 *
 * @param color the hexadecimal color to modify
 * @param percent the shade value as a percentage
 */
function shadeColor(color: string, percent: number) {
  const hex = color.replace('#', '');
  const r = Math.round((parseInt(hex.substring(0, 2), 16) * (100 + percent)) / 100);
  const g = Math.round((parseInt(hex.substring(2, 4), 16) * (100 + percent)) / 100);
  const b = Math.round((parseInt(hex.substring(4, 6), 16) * (100 + percent)) / 100);
  const rHex = (r <= 255 ? r : 255).toString(16).padStart(2, '0');
  const gHex = (g <= 255 ? g : 255).toString(16).padStart(2, '0');
  const bHex = (b <= 255 ? b : 255).toString(16).padStart(2, '0');
  return `#${rHex}${gHex}${bHex}`;
}

export { shadeColor };
