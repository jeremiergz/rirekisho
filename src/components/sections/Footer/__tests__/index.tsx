import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from 'theme';
import Footer from '..';
import mockData from './mock.json';

describe('components/sections/Footer Test Suite', () => {
  beforeAll(() => {
    const graphql = jest.spyOn(Gatsby, 'graphql');
    graphql.mockImplementation(() => '');
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => mockData);
  });

  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
