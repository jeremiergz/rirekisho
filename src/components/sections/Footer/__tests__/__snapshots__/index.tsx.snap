// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sections/Footer Test Suite matches snapshot 1`] = `
<footer
  className="sc-AxjAm gFGULz"
  color="white"
  display="flex"
  height={
    Object {
      "_": 204,
      "tablet": 168,
    }
  }
>
  <div
    className="sc-AxjAm gHjjnZ"
    display="flex"
  >
    <div
      className="sc-AxjAm hVCUTG"
      display="flex"
    >
      <span
        className="sc-AxiKw kJBPsB"
      >
        Made with
      </span>
      <svg
        className="sc-AxhCb cnYhZh"
        fill="#fff"
        height={16}
        viewBox="0 0 455 455"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M455.111 164.089c0 137.956-163.556 228.978-213.333 253.156-8.533 4.267-19.911 4.267-28.444 0C163.556 393.067 0 304.889 0 164.089 0 92.978 52.622 34.667 116.622 34.667c51.2 0 93.867 35.556 109.511 85.333 15.644-49.778 59.733-85.333 109.511-85.333 66.845 0 119.467 58.311 119.467 129.422z"
          fill="#f44336"
        />
        <path
          d="M455.111 164.089c0 137.956-163.556 228.978-213.333 253.156-8.533 4.267-19.911 4.267-29.867 0-22.756-9.956-65.422-34.133-108.089-68.267h1.422c135.111 0 243.2-109.511 243.2-243.2 0-24.178-4.267-48.356-11.378-71.111 66.845 1.422 118.045 58.311 118.045 129.422z"
          fill="#e53935"
        />
        <path
          d="M109.511 142.756c-22.756 5.689-44.089-2.844-48.356-18.489-2.844-17.067 11.378-34.134 34.134-39.823s44.089 2.844 48.356 18.489c4.266 17.067-11.378 34.134-34.134 39.823z"
          fill="#fff"
          opacity="0.2"
        />
      </svg>
      <span
        className="sc-AxiKw kjWeqi"
      >
        using
      </span>
    </div>
    <div
      className="sc-AxjAm jxgjIo"
      display="flex"
    >
      <a
        className="sc-AxjAm sc-AxirZ fwPkvK"
        fontWeight="semi-bold"
        height={64}
        href="https://www.gatsbyjs.org"
        rel="noopener noreferrer"
        target="_blank"
        width={64}
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "maxHeight": null,
              "maxWidth": null,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            aria-hidden={true}
            style={
              Object {
                "paddingBottom": "100%",
                "width": "100%",
              }
            }
          />
          <img
            alt=""
            aria-hidden={true}
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsTAAALEwEAmpwYAAADT0lEQVQ4y5WSWWxMURjHb+9EpBEhqSWRNhE7CeJFKELDtGqKklBStSQay0w7tEwF9WJNiLR31FIGQWutNDzVmtgbkvIgjUTbIF4ExVSnM/dOr983czEz+uIk//y/8y3/833nHEVhleYcU1yZXjGFVcVa2MWuTK3VPfe4uQmw/wA8JY4qWyRu11QrT4lbFClxCXZtAL5XRZlHTOyDFOSBpWAvewNug9Nia4piReWE3449y31JiL0msRXfCGKF4CqoAy58w4g9IafFnX0k2RJLUhIXCTaLt1JoFmV5R8E3yuafM0vmnYqgbP55GbuBnDLxY++SGqdds8V1SLGydlbFb+EWEsvhFTtzL5tOu7cDW8bUnWBzdjX3ecJ0zz0mgs1bc6r+vUeX1TI8EHxB4CU8hpMfljp8UhiUroEI+51RhBndz35IrEai4BDwkcR7iDYwxlBEH5fMO2kWZx2lW+0pGA3kOgrICcBp/wjGzk/gPXBTRIG3ietIw360xXFaBD8DHaFtxFaDH6WOqp6/jlysFTgkY1k+V0Q02umDHbmXZOxPHDIS0XfYvh4fZePsyj8tF2d5+5L8FTywklcRu0PBKK6iHLHhcC3xEPZg6yHjv407uyrxY48HPxD6HPnUdq0Gbox+cO0te3mQKXGT2bW/gl1dXUmBzoBqmqayPqNCtb5SCsU+8E2uIPrK2k9Quz3Plyo5hTMOq5aostkRbUrp7u5WY7tFNCl/0r4/Ps+iE308i6sn0MFYuY6E3AjvLjwb/yCdnZ0poVAow+/397ESbc0v3ve6f72pd6LAZNWpGrqxBHuA+AzDsIXDYZV6BbYpGA6cz0ENuMf4Q2NFfn4PJushY2GC8DMmG9xTt3JCG6LTxdZ1Pbujo6MfbMdfCQqwCyhu5yA7tgeeQHE9sWr2hfAcOpuZnp4uHa4Rwa/WXU4DD0m6QOA2di58NxAIeLCl8/XkvoEl7yaxfCBTnYfr4APwGRG8aITD14LBYBaOx3TrgW/RRQZ8h/tdiV1H54PIvYKvHsH78CJwjQaWwevw6RzWX2lvb+9N4nZdN84QXCzdIpqD7yj7DQiNo2ABvonsD8JTieWCQ+x3Sj7xcrBf6WnJt1H+YyGailgjB6XI/heB0KCWLjFuqAAAAABJRU5ErkJggg=="
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transitionDelay": "500ms",
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              sizes="(max-width: 512px) 100vw, 512px"
              srcSet="/static/eab161f5eea431f46d042aa5def958ab/69585/tech_gatsby.png 200w,
/static/eab161f5eea431f46d042aa5def958ab/497c6/tech_gatsby.png 400w,
/static/eab161f5eea431f46d042aa5def958ab/bc59e/tech_gatsby.png 512w"
            />
            <img
              alt="Go to https://www.gatsbyjs.org"
              loading="lazy"
              onLoad={[Function]}
              sizes="(max-width: 512px) 100vw, 512px"
              src="/static/eab161f5eea431f46d042aa5def958ab/bc59e/tech_gatsby.png"
              srcSet="/static/eab161f5eea431f46d042aa5def958ab/69585/tech_gatsby.png 200w,
/static/eab161f5eea431f46d042aa5def958ab/497c6/tech_gatsby.png 400w,
/static/eab161f5eea431f46d042aa5def958ab/bc59e/tech_gatsby.png 512w"
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 500ms",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><source srcset=\\"/static/eab161f5eea431f46d042aa5def958ab/69585/tech_gatsby.png 200w,
/static/eab161f5eea431f46d042aa5def958ab/497c6/tech_gatsby.png 400w,
/static/eab161f5eea431f46d042aa5def958ab/bc59e/tech_gatsby.png 512w\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" srcset=\\"/static/eab161f5eea431f46d042aa5def958ab/69585/tech_gatsby.png 200w,
/static/eab161f5eea431f46d042aa5def958ab/497c6/tech_gatsby.png 400w,
/static/eab161f5eea431f46d042aa5def958ab/bc59e/tech_gatsby.png 512w\\" src=\\"/static/eab161f5eea431f46d042aa5def958ab/bc59e/tech_gatsby.png\\" alt=\\"Go to https://www.gatsbyjs.org\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </a>
      <a
        className="sc-AxjAm sc-AxirZ fwPkvK"
        fontWeight="semi-bold"
        height={64}
        href="https://styled-components.com"
        rel="noopener noreferrer"
        target="_blank"
        width={64}
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "maxHeight": null,
              "maxWidth": null,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            aria-hidden={true}
            style={
              Object {
                "paddingBottom": "100%",
                "width": "100%",
              }
            }
          />
          <img
            alt=""
            aria-hidden={true}
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAC4jAAAuIwF4pT92AAACrElEQVQ4y2P4//9/GRDb/Pv3zw1Iu2DDQDk4DcNo4iC9VkBcyAAyjIFKAGiWKQPIdBAHSLMCBZhgGMgHYxj/97MvzEAarvnZ7UfMSOpZoQbagwgXKAckwQjCIP6nO5/AGh/OOcl4MHcZM4j9qmiP9Nnqzev7omrlQfxLW04ww/RCaVtkA1ngTv/3G2zo14NTmFC8VHuCY2/x0nWtkaViSN4EOYIZw0CogOz/f/9C//3/z74UKnY05D/HsdqjVYeqT9icazpmsyZ7igNI/M+vPwFA9WbYXOgG5TgDwywVaJgwwgIj8aerao8db2z4f7hm+o19KTNXbwtapw+Su9hzkBmoPgioLwbJQSgGgpJAyr8/X0TA/OthQn/O+Fz5f8b9/7eDtt8ud+T/35I2YRdM8+9fv0WABkYD9fiiG4jsZZl/f3+GgNift6kb/D8JTFpnHf//P2oJxGn/L/XrloDkuhkYQOHmDsTqUH3M2CKFFTkCPm43EP1x0Kz+2RKZll9nQ9v/nzMs+b+KgQ2s9u0kRiRHMOGKZXiy2Ywn8f7dCaGB3oUnMRQDQdkHyUCG3x9ugNgMLzdaM/w54878ZqMJ87ej3sy/TzgxP5wGSVlvd3iA9MEMxpkOwTkBZDMkLf5l/PfrE4T94TDDq0KoF78WMjxbJIWe5ZiwRgqSAmQ2M7IY1EUo6tANdIVygFH5Pw6IgdH6H5Qc/IAYlHg9YTkJFsZIGKuXYQa6AyUjgHQyEPsDcRIQhwGxIa4IgoUjrjCUAWJlIDYGYgOgYkUgrQQV0wRiDSDWBsmBNILkYN7GlfWCgTgfikuAuBmIq6DsPCAuh+JsKD8ZaCkX1OtsUBeDiy8rHIWlIHpiJ6KANQYRhaCSFojtgNgaVIJDvWQBLdZtsGBrqJwtFNtDgyoHAPZcCMWNFSW7AAAAAElFTkSuQmCC"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transitionDelay": "500ms",
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              sizes="(max-width: 512px) 100vw, 512px"
              srcSet="/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/69585/tech_styled_components.png 200w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/497c6/tech_styled_components.png 400w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/bc59e/tech_styled_components.png 512w"
            />
            <img
              alt="Go to https://styled-components.com"
              loading="lazy"
              onLoad={[Function]}
              sizes="(max-width: 512px) 100vw, 512px"
              src="/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/bc59e/tech_styled_components.png"
              srcSet="/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/69585/tech_styled_components.png 200w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/497c6/tech_styled_components.png 400w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/bc59e/tech_styled_components.png 512w"
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 500ms",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><source srcset=\\"/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/69585/tech_styled_components.png 200w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/497c6/tech_styled_components.png 400w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/bc59e/tech_styled_components.png 512w\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 512px) 100vw, 512px\\" srcset=\\"/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/69585/tech_styled_components.png 200w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/497c6/tech_styled_components.png 400w,
/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/bc59e/tech_styled_components.png 512w\\" src=\\"/static/7eb0e9e8f1b57b3d44cb75990ebb50ae/bc59e/tech_styled_components.png\\" alt=\\"Go to https://styled-components.com\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </a>
    </div>
  </div>
  <div
    className="sc-AxjAm bGuxiZ"
    color="caption"
    display="flex"
  >
    <span
      className="sc-AxiKw esNMNr"
    >
      rirekisho v20.07.3
    </span>
    <a
      className="sc-AxjAm sc-AxirZ fMUetU"
      fontWeight="semi-bold"
      href="https://gitlab.com/jeremiergz/rirekisho/tree/20.07.3"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="sc-AxiKw kudcPh"
        fontSize={12}
      >
        source code
      </span>
    </a>
  </div>
</footer>
`;
